app/Http/Controllers/BookController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Book;
+use App\Models\Category;
+use App\Models\Author;
+use App\Models\Publisher;
+use App\Http\Requests\BookRequest;
+use App\Components\Recusive;
+
+class BookController extends Controller
+{
+    private $book;
+    private $category;
+    private $author;
+    private $publisher;
+
+    public function __construct(Book $book, Category $category, Author $author, Publisher $publisher) {
+        $this->book = $book;
+        $this->category = $category;
+        $this->author = $author;
+        $this->publisher = $publisher;
+    } 
+
+    public function index()
+    {
+        $books = Book::latest()->paginate(config('const.five'));
+
+        return view('admin.book.index', compact('books'));
+    }
+
+    public function create()
+    {
+        $categoryOption = $this->getCategory($parentId = '');
```

Comment:
```suggestion
        $categoryOption = $this->getCategory('');
```
tại sao em lại truyền vào rỗng như này?

Reply:
giống hôm qua ạ, truyền rỗng để hàm đệ quy rơi vào trường hợp ko có option nào được selected ạ